server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -408,13 +522,39 @@ private void executePipelines(
                     throw new IllegalArgumentException("pipeline with id [" + pipelineId + "] does not exist");
                 }
                 Pipeline pipeline = holder.pipeline;
+                String oldIndex = indexRequest.indices()[0];
```

Comment:
maybe rename `oldIndex` to `originalIndex`?

Reply:
Done